test/unit/realm_test.rb

```diff
@@ -8,42 +8,30 @@ def setup
   end
 
   test "should not save without a name" do
-    assert !@new_realm.save
+    assert validate_presence_of(:name)
   end
 
   test "should exists a unique name" do
-    other_realm = Realm.new(:name => "myrealm.net")
-    assert !other_realm.save
+    assert validate_uniqueness_of(:name)
   end
 
   test "when cast to string should return the name" do
-    s = @realm.to_s
-    assert_equal @realm.name, s
+    assert_equal @realm.name, @realm.to_s
   end
 
   test "should not destroy if it contains hosts" do
-    disable_orchestration
-    host = FactoryGirl.create(:host, :realm => @realm)
-
-    assert host.save
-
-    realm = host.realm
-    assert !realm.destroy
-    assert_match /is used by/, realm.errors.full_messages.join("\n")
+    assert have_many(:hosts).
+      class_name('Host::Managed').
+      dependent(:destroy)
   end
 
   test "realm can be assigned to locations" do
-    location1 = Location.create :name => "Zurich"
-    assert location1.save!
-
-    location2 = Location.create :name => "Switzerland"
-    assert location2.save!
-
-    realm = Realm.create :name => "test.net", :realm_proxy => smart_proxies(:realm), :realm_type => "FreeIPA"
-    realm.locations.destroy_all
-    realm.locations.push location1
-    realm.locations.push location2
-    assert realm.save!
+    assert have_many(:locations).
+      through(:taxable_taxonomies).
```

Comment:
I'd suggest putting a shortcut into test_helper for this, as it'll probably be useful elsewhere.